convert2rhel/unit_tests/hostmetering_test.py

```diff
@@ -0,0 +1,130 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__metaclass__ = type
+
+import pytest
+import six
+
+from convert2rhel import hostmetering, utils
+from convert2rhel.systeminfo import Version, system_info
+from convert2rhel.unit_tests import RunSubprocessMocked
+
+
+six.add_move(six.MovedModule("mock", "mock", "unittest.mock"))
+from six.moves import mock
+
+
+@pytest.mark.parametrize(
+    ("rhsm_facts", "os_version", "should_configure_metering", "envvar"),
+    (
+        (
+            {},
+            Version(7, 9),
+            False,  # not on hyperscaller
+            "auto",
+        ),
+        (
+            {"aws_instance_id": "i-1234567890abcdef0"},
+            Version(7, 9),
+            True,
+            "auto",
+        ),
+        (
+            {"azure_instance_id": "012345678-abcde-efgh-1234-abcdefgh1234"},
+            Version(7, 9),
+            True,
+            "auto",
+        ),
+        (
+            {"gcp_instance_id": "12345-6789-abcd-efgh-0123456789ab"},
+            Version(7, 9),
+            True,
+            "auto",
+        ),
+        (
+            {"aws_instance_id": "i-1234567890abcdef0"},
+            Version(8, 8),
+            False,  # not on RHEL 7
+            "auto",
+        ),
+        (
+            {"azure_instance_id": "012345678-abcde-efgh-1234-abcdefgh1234"},
+            Version(8, 8),
+            False,  # not on RHEL 7
+            "auto",
+        ),
+        (
+            {"gcp_instance_id": "12345-6789-abcd-efgh-0123456789ab"},
+            Version(8, 8),
+            False,  # not on RHEL 7
+            "auto",
+        ),
+        (
+            {},
+            Version(7, 9),
+            True,  # forced
+            "force",
+        ),
+        (
+            {"aws_instance_id": "i-1234567890abcdef0"},
+            Version(8, 8),
+            True,  # forced
+            "force",
+        ),
+        (
+            {"aws_instance_id": "i-1234567890abcdef0"},
+            Version(7, 9),
+            False,
+            "arbitrary",  # unknown option
+        ),
+        (
+            {"aws_instance_id": "i-1234567890abcdef0"},
+            Version(7, 9),
+            False,
+            "",  # option left empty
+        ),
+    ),
+)
+def test_hostmetering(monkeypatch, rhsm_facts, os_version, should_configure_metering, envvar):
+    if envvar is not None:
+        monkeypatch.setenv("CONVERT2RHEL_CONFIGURE_HOST_METERING", envvar)
+
+    monkeypatch.setattr(system_info, "version", os_version)
+    monkeypatch.setattr(system_info, "releasever", "")  # reset as other test set it
```

Comment:
If other test set it globally, we should update the test to avoid test pollution down the road. 

We should fix as much test pollution as possible.

Reply:
Your unit_test should not reset the releasever because another test populated it wrongly, but rather, we should fix the test that pollutes it

Reply:
I commented it out and the tests still pass. @pvoborni do you know what test is being referred to here?

Reply:
IIRC (was in December), I got a test failure without it. But unfortunately I don't remember more. So if it passes without then it can be removed.

Reply:
Possibly it was another test pollution we had, but maybe got solved with the recent merges then. I fixed a couple of them in some of my PRs that got merged.